src/Twig/RolesMatrixExtension.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Twig;
+
+use Sonata\UserBundle\Security\RolesBuilder\RolesBuilderInterface;
+use Symfony\Component\Form\FormView;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ * @author Cengizhan Çalışkan <cengizhancaliskan@gmail.com>
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+final class RolesMatrixExtension extends AbstractExtension
+{
+    /**
+     * @var RolesBuilderInterface
+     */
+    private $rolesBuilder;
+
+    public function __construct(RolesBuilderInterface $rolesBuilder)
+    {
+        $this->rolesBuilder = $rolesBuilder;
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('renderMatrix', [$this, 'renderMatrix'], ['needs_environment' => true]),
+            new TwigFunction('renderRolesList', [$this, 'renderRolesList'],
+                ['needs_environment' => true]),
+        ];
+    }
+
+    public function getName(): string
+    {
+        return self::class;
+    }
+
+    public function renderRolesList(Environment $environment, FormView $form): ?string
```

Comment:
Can this really return `null`?

Reply:
Test says yea

Reply:
Isn't that because how $environment is mocked?